File: python/tracing/openai-agents-sdk/verification_mcp_server.py
"""Local deterministic MCP tool used by latest SDK verification."""

from mcp.server.mcpserver import MCPServer

server = MCPServer("verification-weather")


@server.tool()
def forecast_mcp(city: str) -> str:
    """Return the verified weather for a city."""
    return f"Sunny in {city}; 22 C."


if __name__ == "__main__":
    server.run(transport="stdio")

File: typescript/tracing/openai-agents-sdk/verification-mcp-server.mjs
import { McpServer } from '@modelcontextprotocol/server';
import { StdioServerTransport } from '@modelcontextprotocol/server/stdio';
import { z } from 'zod';
const server = new McpServer({name:'verification-weather',version:'1.0.0'});
server.registerTool('forecast_mcp', {description:'Return the verified weather for a city.',inputSchema:{city:z.string()}},
  async({city})=>({content:[{type:'text',text:`Sunny in ${city}; 22 C.`}]}));
await server.connect(new StdioServerTransport());
